detectron2/export/__init__.py

```diff
@@ -9,7 +14,13 @@
 else:
     from .api import *
 
-from .flatten import TracingAdapter
-from .torchscript import scripting_with_instances, dump_torchscript_IR
+
+STABLE_ONNX_OPSET_VERSION = 11
+
+
+def add_export_config(cfg):
+    warnings.warn("add_export_config has been deprecated and behaves as no-op function.")
+    return cfg
+
 
```

Comment:
This is still a public API and fully working for detectron2 + pytorch < 1.11, so better give users a grace period before removing it. What do you think?

Reply:
I'm not suggesting removing it. But I suggest to keep the function where it was and it seems it won't break anyone. So there is no need to move the function into `__init__.py`

Reply:
For PyTorch >= 1.11, the current `detectron2` implementation (aka keeping `add_export_config` at `detectron2/export/api.py`) does break existing training scripts. See a full repro below:

```bash
conda create -n torch111_py39cu113 python=3.9 numpy
conda activate torch111_py39cu113 
conda install pytorch torchvision cudatoolkit=11.3 -c pytorch # pytorch 1.11
python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'
python -c "from detectron2.export import add_export_config"
```
The result is:
```bash
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name 'add_export_config' from 'detectron2.export' (/anaconda3/envs/torch111_py39cu113/lib/python3.9/site-packages/detectron2/export/__init__.py)
```
The root cause is that after a "recent" [PyTorch BC change](https://github.com/pytorch/pytorch/pull/66658), `libcaffe2` is no longer distributed along with official PyTorch packages. When `detectron2` tries to import anything from `detectron2.export.apy` namespace, it will try to import caffe2 stuff indirectly, such as `from caffe2.proto import caffe2_pb2`

This change tries to create an *explicit* deprecation path by printing a scary warning so that users update their code while it still keeps things working (by moving `add_export_config` to a caffe2-free spot)